src/components/file/file.service.ts

```diff
@@ -138,13 +138,18 @@ export class FileService {
     session: ISession
   ): Promise<FileListOutput> {
     const result = await this.repo.getChildrenById(session, parentId, input);
-    const items = await Promise.all(
+    let items = await Promise.all(
       result.children.map((node) => this.adaptBaseNodeToFileNode(node, session))
     );
+    if (input?.filter?.category)
+      items = items.filter((item) =>
+        input.filter.category!.includes(item.category)
+      );
```

Comment:
Filters need to be applied in the query, otherwise `total` and pagination is invalid